### DISCUSSION TITLE
Power Script Support

### DISCUSSION THREAD

**Discussion by stueble** [GioF71/squeezelite-docker#317] [Ideas]
Hi,

I noticed that squeezelite-docker support a huge number of command line options of squeezelite, but unfortunately not "-S", the power scripts (e.g., to enable/disable an amp). Any chance to add that parameter? 

The required script can be mounted in the docker by the user.

Best regards,
Chris

**Comment by GioF71** [comment]
Hello, yes, I will do that. Keep in mind that the script will run on the container, so if you want to execute something that has an effect on the host device, you will need to use this script in a way that it can trigger an action on the host.

**Comment by GioF71** [comment]
Hello, there is a new set of images building now.
We have support for -S and -G.
Not every version of squeezelite seems to support -S. The binaries from sourceforge have support for the feature, so you might want to select the sourceforge variants.
Let me know if everything is ok.

Regards
Giovanni

**Reply by stueble** [reply]
Many thanks, I will try!

**Comment by vojtechj** [comment]
Hi, what is it good for? does the bluetooth container support it?
I am asking, because I would like to know if it is possible to signal that the bluetooth speakers were turned on and off. I can see that in the container logs, that the playback started/stoped when the bluetooth speakers were turned on/off.
Thank you

**Comment by GioF71** [comment]
Hello, this is a request from a user, it is available in all the container images. Keep in mind that the script will run in the container, so if you want to execute some actions on the host machine, you need to get creative and imagine a method to trigger such desired actions. Maybe you can write to a file in a mounted volume, and have a process in the host machine looking for changes on that file maybe. Just saying.
What would you like to implement exactly?

**Reply by vojtechj** [reply]
I am playing radio stream from home assistant via slimproto integration and with the help of squeezelite and bluealsa I am "streaming" it to a panasonic sc-hc200 radio. But when I turn of the sc-hc200 the home assistant doesn't get it a is still playing that online radio stream. But I saw in the container logs, that bluealsa stops "playing" when I turn of sc-hc200. So I am curious if it is possible to notice home assistant about bluetooth device got turned off so home assistant should stop the playback or when it turned on so home assistant can start the playback.

**Reply by GioF71** [reply]
Hello, can't home assistant connect to bluetooth devices directly?

**Reply by vojtechj** [reply]
to bluetooth devices yes, but not to bluetooth speakers

**Reply by vojtechj** [reply]
https://www.reddit.com/r/homeassistant/comments/1ej2rcs/any_way_to_use_a_bluetooth_speaker_in_ha/

after connecting the bluetooth speaker directly from raspbian I had to disable bluetooth in home assistant, because the audio playback was choppy .... probably because of HA polling on the bluetooth module for new devices/device changes.

**Reply by GioF71** [reply]
OK, I did not know about that and I assumed that HA could play to BT speakers.
Well considering the complexity, I would try to make squeezelite work in your setup on "bare-metal" first, then if everything works, I would try to use the docker container, just to avoid to have to deal with container-related issues upfront, like for example, having to call a script that might require additional dependencies, etc.
Maybe you could install bluez-alsa on the host device, and create systemd unit maybe? I might be able to help you with that.

**Reply by vojtechj** [reply]
ok, thank you.
I will look into that and see if I can figure out something.

**Reply by vojtechj** [reply]
Just as a quick try, I loaded up pi zero 2 with picoreplayer and it works with home assistant as it should. when I turn off the panasonic radio the device in HA becomes unavailable right away and when I turn it on, it connects to BT and in HA it becomes available again.

**Reply by GioF71** [reply]
Hello, I was about to suggest picorePlayer as it has the built in feature of bt support, but I did not know if you wanted to run other players/containers on the same device. Anyway I am happy that pcp works for you. Is that already the solution or just a step forward?

P.s. I am also starting to use home assistant and especially music assistant. On the latter I like to use slimproto players for the best functionality but then I lose last.fm support which works with standard home assistant players. There's a lot of potential though!

**Reply by vojtechj** [reply]
HI, it's just a temporary solution, because I am running HA in container and esphome as well and some other containers. And I don't want to "waste" another device or consume more energy than needed, because I can't find a solution yet.
But it's a proof, that squeezelite can send BT device availability ... I just have to figure out how or find out why it doesn't work with your bluetooth version of squeezelite container.

**Reply by GioF71** [reply]
Hello, I share your thoughts. I also think containerization is great for reducing the hardware requirements, otherwise as you say you need a Pi for each "thing" you want to do, and that's a waste. Honestly I have to say that I don't own a lot of bt speakers, the main one is my wife's and she uses it directly with her phone with Spotify.
I also have a small soundbar for the (dumb) kitchen tv but it's mainly used via an optical cable to an android tv box, so I do not use it with bt very often, to say the least.
But I would like to buy one for a smart-working desk I have at home, as I really don't like the sound that comes out of the monitor speakers, and I'd prefer to avoid buying yet another set of dac, amplifier and speakers, also because of space limitations.

**Reply by GioF71** [reply]
Hello, I have done some testing with the Anker device my wife usually uses. I think I might have a decent solution.

First step is of course to do the usual trust/pair/connect process.

Then I installed [this application](https://github.com/jrouleau/bluetooth-autoconnect) via a systemd unit. It handles automatic reconnection. Great!

I then created a systemd unit for the squeezelite at /etc/systemd/system/squeezelite-anker.service:

```text
[Unit]
Description=Squeezelite (DietPi)
Documentation=man:squeezelite(1) https://ralph-irving.github.io/squeezelite.html
Wants=network-online.target
After=network-online.target sound.target

[Service]
User=squeezelite
# EnvironmentFile=/etc/default/squeezelite
ExecStart=/usr/bin/squeezelite -n AnkerBT -o "bluealsa:DEV=F4:2B:7D:23:E1:9C,PROFILE=a2dp,SRV=org.bluealsa" -s 192.168.1.174

[Install]
WantedBy=multi-user.target
```

Of course "F4:2B:7D:23:E1:9C" is my bt device and 192.168.1.174 is my Lyrion Music Server.

When I turn off the device, the squeezelite player does not disappear from LMS unfortunately. I suppose for that we might need a program that starts the player when the device connects, and stops it when it disconnects. Squeezelite remains running if the bt device disconnects.

However, the player can even resume playing if I power-cycle the speaker.

Do you think this might help?
Next step: running it in docker. But not mandatory I believe :-)

**Reply by vojtechj** [reply]
Hi, thank you ... I am currently looking at how it's made in PiCorePlayer, but I have very little spare time for that. But I will report back if I find out something.
